### ISSUE TITLE
Use a code formatter

### ISSUE THREAD

**Issue by cosimoNigro** [javierrico/gLike#26]
The code is not properly formatted at the moment. 
C++ easily develop many nested for loops and the misuse of indentation and spaces can cause visualisation problems with different text editors.

Here for example a comparison of the same file open with `vim` (left) and `sublimetext` (right).
As you can see in the second case the lines in the `if` control blocks result indented on the left, rather than on the right.

<img width="1402" alt="Screenshot 2020-02-03 at 09 42 13" src="https://user-images.githubusercontent.com/26409711/73637909-9dbad900-4669-11ea-8033-cf210bab2cd4.png">

Using a code formatter will improve gLike appearance, readability and modification. 

I have used [clang-format](https://clang.llvm.org/docs/ClangFormatStyleOptions.html) in the past.
If we decide to use it we can establish which coding [style](https://clang.llvm.org/docs/ClangFormatStyleOptions.html) and indentation type (level) we want, then before each pull request we just have to remember to launch e.g.

```bash
clang-format -style="{BasedOnStyle: llvm, IndentWidth: 4, UseTab: Never}" -i src/Iact1dUnbinnedLkl.cc
```

It will uniform the loops, and make consistent tabs and spacings.
Do you think is it worth considering it? It will simplify our lives.

**Comment by javierrico** [comment]
Yes, I like that, but please define the "right" [i.e. mine] style!
